File: AppointmentService/src/model/Appointment.java
package model;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.Statement;

public class Appointment {
	
	public Connection connect() {
		Connection con = null;
		try {
			Class.forName("com.mysql.jdbc.Driver");
			con = (Connection) DriverManager.getConnection("jdbc:mysql://127.0.0.1:3306/appointmentdb", "root", "");
			// For testing
			System.out.print("Successfully connected");
		} catch (Exception e) {
			e.printStackTrace();
		}
		return con;
	}

	public String insertApointment(String doctor_id, String patient_id, String prescription, String doc_notes) {

		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database";
			}
			// create a prepared statement
			String query = "insert into appointmentdetails(app_id, doctor_id, patient_id, prescription, doc_notes)"
					+ " values ( ?,  ?,  ?,  ?,  ?)";
			PreparedStatement preparedStmt = con.prepareStatement(query);


			// binding values
			preparedStmt.setInt(1, 0);
			preparedStmt.setString(2, doctor_id);
			preparedStmt.setString(3, patient_id);
			preparedStmt.setString(4, prescription);
			preparedStmt.setString(5, doc_notes);
			
			
			// execute the statement    
			preparedStmt.execute();    
			con.close(); 
	   
			String newAppo = readAppo(); 
			output =  "{\"status\":\"success\", \"data\": \""
			+ newAppo + "\"}";    
		}   
		catch (Exception e)   
		{    
			output =  "{\"status\":\"error\", \"data\": \"Error while inserting the Appointment.\"}";  
			System.err.println(e.getMessage());   
		} 
		
	  return output;  
	} 
	
	public String readAppo()  

	{   
		String output = ""; 
	
		try   
		{    
			Connection con = connect(); 
		
			if (con == null)    
			{
				return "Error while connecting to the database for reading."; 
			} 
	 
			// Prepare the html table to be displayed    
			output = "<table border='1'><tr><th>Doctor Id</th><th>Patient Id</th><th>Prescription</th><th>Doctor Notes</th><th>Update</th><th>Remove</th></tr>"; 
	 
			String query = "select * from appointmentdetails";    
			Statement stmt = con.createStatement();    
			ResultSet rs = stmt.executeQuery(query); 
	 
			// iterate through the rows in the result set    
			while (rs.next())    
			{     
				String app_id = Integer.toString(rs.getInt("app_id"));
				String doctor_id = rs.getString("doctor_id");
				String patient_id = rs.getString("patient_id");
				String prescription = rs.getString("prescription");
				String doc_notes = rs.getString("doc_notes");
			
	 
				// Add into the html table 
				output += "<tr><td><input id=\'hidItemIDUpdate\' name=\'hidItemIDUpdate\' type=\'hidden\' value=\'" + app_id + "'>" 
							+ doctor_id + "</td>";      
				output += "<td>" + patient_id + "</td>";     
				output += "<td>" + prescription + "</td>";
				output += "<td>" + doc_notes + "</td>"; 
	 
				// buttons     
//				output += "<td><input name=\'btnUpdate\' type=\'button\' value=\'Update\' class=\' btnUpdate btn btn-secondary\'></td>"
//						//+ "<td><form method=\"post\" action=\"items.jsp\">      "
//						+ "<input name=\'btnRemove\' type=\'submit\' value=\'Remove\' class=\'btn btn-danger\'> "
//						+ "<input name=\"hidItemIDDelete\" type=\"hidden\" value=\"" + itemID + "\">" + "</form></td></tr>"; 
				output +="<td><input name='btnUpdate' type='button' value='Update' class='btnUpdate btn btn-secondary'></td>"       
							+ "<td><input name='btnRemove' type='button' value='Remove' class='btnRemove btn btn-danger' data-docid='" + app_id + "'>" + "</td></tr>"; 
			} 
	 
			con.close(); 
	 
			// Complete the html table    
			output += "</table>";   
		}   
		catch (Exception e)   
		{    
			output = "Error while reading the Appointment.";    
			System.err.println(e.getMessage());   
		} 
	 
		return output;  
	}
	 
	
	public String updateAppointment(String app_id, String doctor_id, String patient_id, String prescription, String doc_notes) {
		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for updating.";
			}
			// create a prepared statement
			String query = "UPDATE appointmentdetails SET doctor_id=?,patient_id=?,prescription=?,doc_notes=? WHERE app_id=?";
			PreparedStatement preparedStmt = con.prepareStatement(query);
			// binding values
			preparedStmt.setString(1, doctor_id);
			preparedStmt.setString(2, patient_id);
			preparedStmt.setString(3, prescription);
			preparedStmt.setString(4, doc_notes);
			preparedStmt.setInt(5, Integer.parseInt(app_id));	 
			
			// execute the statement    
			preparedStmt.execute();    
			con.close(); 
	 
			String newAppo = readAppo();    
			output = "{\"status\":\"success\", \"data\": \"" +        
					newAppo + "\"}";    
		}   
		catch (Exception e)   
		{    
			output =  "{\"status\":\"error\", \"data\": \"Error while updating the item.\"}";   
			System.err.println(e.getMessage());   
		} 
	 
	  return output;  
	} 
	
	public String deleteAppor(String app_id) {
		String output = "";
		try {
			Connection con = connect();
			if (con == null) {
				return "Error while connecting to the database for deleting.";
			}
			// create a prepared statement
			String query = "delete from appointmentdetails where app_id=?";
			PreparedStatement preparedStmt = con.prepareStatement(query);
			// binding values
			preparedStmt.setInt(1, Integer.parseInt(app_id));
	 
			// execute the statement    
			preparedStmt.execute();    
			con.close(); 
	 
			String newAppo = readAppo();    
			output = "{\"status\":\"success\", \"data\": \"" +       
					newAppo + "\"}";    
		}   
		catch (Exception e)   
		{    
			output = "Error while deleting the Appointment.";    
			System.err.println(e.getMessage());   
		} 
	 
		return output;  
	}
	 
	 
}

File: AppointmentService/WebContent/Components/appointment.js
$(document).ready(function() 
{  
	if ($("#alertSuccess").text().trim() == "")  
	{   
		$("#alertSuccess").hide();  
	}  
	$("#alertError").hide(); }); 
 
// SAVE ============================================ 
$(document).on("click", "#btnSave", function(event) 
{  
	// Clear alerts---------------------  
	$("#alertSuccess").text("");  
	$("#alertSuccess").hide();  
	$("#alertError").text("");  
	$("#alertError").hide(); 
 
	// Form validation-------------------  
	var status = validateItemForm();  
	if (status != true)  
	{   
		$("#alertError").text(status);   
		$("#alertError").show();   
		return;  
	} 
 
	// If valid------------------------  
	var type = ($("#hidItemIDSave").val() == "") ? "POST" : "PUT"; 
	
	$.ajax( 
	{  
		url : "AppointmentAPI",  
		type : type,  
		data : $("#formItem").serialize(),  
		dataType : "text",  
		complete : function(response, status)  
		{   
			onItemSaveComplete(response.responseText, status);  
		} 
	}); 
}); 

function onItemSaveComplete(response, status) 
{  
	if (status == "success")  
	{   
		var resultSet = JSON.parse(response); 

		if (resultSet.status.trim() == "success")   
		{    
			$("#alertSuccess").text("Successfully saved.");    
			$("#alertSuccess").show(); 

			$("#divItemsGrid").html(resultSet.data);   
		} else if (resultSet.status.trim() == "error")   
		{    
			$("#alertError").text(resultSet.data);    
			$("#alertError").show();   
		} 

	} else if (status == "error")  
	{   
		$("#alertError").text("Error while saving.");   
		$("#alertError").show();  
	} else  
	{   
		$("#alertError").text("Unknown error while saving..");   
		$("#alertError").show();  
	} 

	$("#hidItemIDSave").val("");  
	$("#formItem")[0].reset(); 
} 
 
// UPDATE========================================== 
$(document).on("click", ".btnUpdate", function(event) 
{     
	$("#hidItemIDSave").val($(this).closest("tr").find('#hidItemIDUpdate').val());     
	$("#doctor_id").val($(this).closest("tr").find('td:eq(0)').text());     
	$("#patient_id").val($(this).closest("tr").find('td:eq(1)').text());     
	$("#prescription").val($(this).closest("tr").find('td:eq(2)').text());
	$("#doc_notes").val($(this).closest("tr").find('td:eq(3)').text()); 
}); 

//REMOVE===========================================
$(document).on("click", ".btnRemove", function(event) 
{  
	$.ajax(  
	{   
		url : "AppointmentAPI",   
		type : "DELETE",   
		data : "app_id=" + $(this).data("app_id"),   
		dataType : "text",   
		complete : function(response, status)   
		{    
			onItemDeleteComplete(response.responseText, status);   
		}  
	}); 
}); 

function onItemDeleteComplete(response, status) 
{  
	if (status == "success")  
	{   
		var resultSet = JSON.parse(response); 

		if (resultSet.status.trim() == "success")   
		{    
			$("#alertSuccess").text("Successfully deleted.");    
			$("#alertSuccess").show(); 
		
			$("#divItemsGrid").html(resultSet.data);   
		} else if (resultSet.status.trim() == "error")   
		{    
			$("#alertError").text(resultSet.data);    
			$("#alertError").show();   
		}

	} else if (status == "error")  
	{   
		$("#alertError").text("Error while deleting.");   
		$("#alertError").show();  
	} else  
	{   
		$("#alertError").text("Unknown error while deleting..");   
		$("#alertError").show();  
	}
}

// CLIENT-MODEL========================================================================= 
function validateItemForm() 
{  
	// doctor id  
	if ($("#doctor_id").val().trim() == "")  
	{   
		return "Insert Doctor ID.";  
	} 
 
	// patient id  
	if ($("#patient_id").val().trim() == "")  
	{   
		return "Insert patient Id.";  
	} 
	// prescription  
	if ($("#prescription").val().trim() == "")  
	{   
		return "Insert Doctor prescription.";  
	}   
	
	// doc notes  
	if ($("#doc_notes").val().trim() == "")  
	{   
		return "Insert Doctor Notes.";  
	} 


	return true; 
}